app/src/main/java/edu/wpi/first/shuffleboard/app/MainWindowController.java

```diff
@@ -138,6 +127,121 @@ private void initialize() throws IOException {
+        pluginStage.close();
+      }
+    });
+    pluginStage.setScene(new Scene(pluginPane));
+    pluginStage.sizeToScene();
+    pluginStage.setMinWidth(675);
+    pluginStage.setMinHeight(325);
+    pluginStage.setTitle("Loaded Plugins");
+    EasyBind.listBind(pluginPane.getStylesheets(), root.getStylesheets());
+  }
+
+  /**
+   * Sets up UI components to represent the sources that a plugin defines.
+   */
+  private void setup(Plugin plugin) {
+    plugin.getSourceTypes().forEach(sourceType -> {
+      SourceTreeTable<SourceEntry<?>, ?> tree = new SourceTreeTable<>();
+      tree.setSourceType(sourceType);
+      tree.setRoot(new TreeItem<SourceEntry<?>>(sourceType.createRootSourceEntry()));
+      tree.setShowRoot(false);
+      tree.setSortPolicy(__ -> {
+        sortTree(tree.getRoot());
+        return true;
+      });
+      tree.getSelectionModel().selectedItemProperty().addListener((__, oldItem, newItem) -> {
+        selectedEntry = newItem == null ? null : newItem.getValue();
+      });
+      tree.setRowFactory(__ -> {
+        TreeTableRow<SourceEntry<?>> row = new TreeTableRow<>();
+        makeSourceRowDraggable(row);
+        return row;
+      });
+      tree.setOnContextMenuRequested(e -> {
+        TreeItem<SourceEntry<?>> selectedItem = tree.getSelectionModel().getSelectedItem();
+        if (selectedItem == null) {
+          return;
+        }
+
+        DataSource<?> source = selectedItem.getValue().get();
+        List<String> widgetNames = Widgets.getDefault().widgetNamesForSource(source);
+        if (widgetNames.isEmpty()) {
+          // No known widgets that can show this data
+          return;
+        }
+
+        ContextMenu menu = new ContextMenu();
+        widgetNames.stream()
+            .map(name -> createShowAsMenuItem(name, source))
+            .forEach(menu.getItems()::add);
+
+        menu.show(tree.getScene().getWindow(), e.getScreenX(), e.getScreenY());
+      });
+      sourceType.getAvailableSources().addListener((MapChangeListener<String, Object>) change -> {
+        SourceEntry entry = sourceType.createSourceEntryForUri(change.getKey());
+        if (change.wasAdded()) {
+          tree.updateEntry(entry);
+        } else if (change.wasRemoved()) {
+          tree.removeEntry(entry);
+        }
+      });
+      sourceType.getAvailableSourceUris().stream()
+          .map(sourceType::createSourceEntryForUri)
+          .forEach(tree::updateEntry);
+      TitledPane titledPane = new TitledPane(sourceType.getName(), tree);
+      sourcesAccordion.getPanes().add(titledPane);
+      sourcesAccordion.setExpandedPane(titledPane);
+    });
+
+    // Add widgets to the gallery as well
+    widgetGallery.setWidgets(Widgets.getDefault().allWidgets());
+  }
+
+  /**
+   * Removes all traces from a plugin from the application window. Source trees will be removed and all widgets
+   * defined by the plugin will be removed from all dashboard tabs.
+   */
+  private void tearDown(Plugin plugin) {
```

Comment:
Bind widgetGallery how? We'd have to pass it into the plugin loader when the controller is initialized